### PR TITLE
CLI development improvements

### PR DESCRIPTION
**Pull request by bowlofeggs**
This pull request started off as a simple idea - make the ```bodhi``` client in the development environment use the Bodhi server from the development environment. Unfortunately, the CLI's code had very low test coverage and so I had to write a lot of new test classes to cover the new code I needed to add. As a result, this pull request raises the test coverage by another percentage point!

There are five commits in this pull request, as they build upon each other but are atomic:

1. Modify the bindings so they will handle the Bodhi base_url not ending in a slash (for human friendliness).
2. Add some docblocks to the client code.
3. Add a new ```--url``` flag to the CLI, test coverage for the new flag, and documentation for the new flag.
4. Configure the development environment to use the new ```BODHI_URL``` environment variable by default, which was the ultimate goal of the pull request.
5. Require test coverage to be at least 82%, since this pull request raised the coverage by another percentage.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -2,7 +2,7 @@
 cover-erase=TRUE
 cover-html=TRUE
 cover-inclusive=TRUE
-cover-min-percentage=81
+cover-min-percentage=82
```
Comment (jeremycline) [inline setup.cfg]: Aw yiss

**Comment by jeremycline** [comment]
I see a few failing unit tests:
```
======================================================================
FAIL: Assert that a successful operation is handled properly.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/home/vagrant/bodhi/bodhi/tests/client/test___init__.py", line 216, in test_successful_operation
    password='s3kr3t', staging=False)
  File "/usr/lib/python2.7/site-packages/mock/mock.py", line 948, in assert_called_once_with
    return self.assert_called_with(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "/usr/lib/python2.7/site-packages/six.py", line 718, in raise_from
    raise value
AssertionError: Expected call: __init__(base_url='https://bodhi.fedoraproject.org/', password='s3kr3t', staging=False, username='some_user')
Actual call: __init__(base_url=u'http://localhost:6543/', password=u's3kr3t', staging=False, username=u'some_user')

======================================================================
FAIL: Assert that request() transforms a bodhi.client.bindings.UpdateNotFound into a
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/home/vagrant/bodhi/bodhi/tests/client/test___init__.py", line 245, in test_update_not_found
    password='s3kr3t', staging=False)
  File "/usr/lib/python2.7/site-packages/mock/mock.py", line 948, in assert_called_once_with
    return self.assert_called_with(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "/usr/lib/python2.7/site-packages/six.py", line 718, in raise_from
    raise value
AssertionError: Expected call: __init__(base_url='https://bodhi.fedoraproject.org/', password='s3kr3t', staging=False, username='some_user')
Actual call: __init__(base_url=u'http://localhost:6543/', password=u's3kr3t', staging=False, username=u'some_user')
```
for example. 

I'm guessing [BASE_URL](https://github.com/fedora-infra/bodhi/pull/1314/files#diff-48a1af2942ae702129b3862397254f6aR244) isn't right in the bindings?

**Comment by bowlofeggs** [comment]
@jeremycline Ah yes I see those now. I think I had run the test suite without the new ```BODHI_URL``` environment variable set. When that is set, it doesn't pass as you mention above. I'll rework this so it passes with or without that set.

**Comment by bowlofeggs** [comment]
@jeremycline I think I have the tests passing now, whether or not that environment variable is set. I did it by having the tests look at the same environment variable and deciding whether to use it or the default.

**Comment by jeremycline** [comment]
I'm still seeing tests failing 😢 

I'm doing a fresh ``vagrant up`` on this branch and running btest:

```
======================================================================
FAIL: The Update.days_to_stable() method should return a positive integer if
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/vagrant/bodhi/bodhi/tests/server/test_models.py", line 315, in test_days_to_stable_not_meets_testing_requirements_with_date_testing
    eq_(update.days_to_stable, 3)
AssertionError: 0 != 3
```

```
======================================================================
FAIL: test_edit_update (bodhi.tests.server.services.test_updates.TestUpdatesService)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/home/vagrant/bodhi/bodhi/tests/server/services/test_updates.py", line 1645, in test_edit_update
    self.assertMultiLineEqual(up['comments'][-1]['text'], comment)
AssertionError: u'guest edited this update.\n\nNew build(s):\n\n- bodhi-2.0.0-3.fc17\n\nRemoved  [truncated]... != 'guest edited this update.\n\nNew build(s):\n\n- bodhi-2.0.0-3.fc17\n\nRemoved b [truncated]...
  guest edited this update.
  
  New build(s):
  
  - bodhi-2.0.0-3.fc17
  
  Removed build(s):
  
- - bodhi-2.0.0-2.fc17+ - bodhi-2.0.0-2.fc17
?                     +
+ 
+ Karma has been reset.
```

```
======================================================================
ERROR: The Update.days_to_stable() method should return 0 if Update.meets_testing_requirements()
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/vagrant/bodhi/bodhi/tests/server/test_models.py", line 290, in test_days_to_stable_meets_testing_requirements
    eq_(update.days_to_stable, 0)
  File "/home/vagrant/bodhi/bodhi/server/models/models.py", line 1862, in days_to_stable
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'NoneType'
```

For what it's worth, these appear to also be failing on the develop branch.

**Comment by bowlofeggs** [comment]
@jeremycline I'm not able to replicate these failing tests, but you definitely have at least one stale .pyc on your system, because that last traceback you cited has a now-nonexisting ```bodhi/bodhi/server/models/models.py``` file.

Can you try again after ```find . -name "*.pyc" -delete```?

Also, this PR is constantly needing rebasing because it keeps conflicting. I think I've already rebased it twice! Unfortunately I don't have time to do that tonight because I'm about to go away for the weekend ☹

**Comment by bowlofeggs** [comment]
P.S. Maybe we should switch to tox and have it rm the .pyc files… Or we could have btest do that.

**Review by jeremycline — APPROVED** [review]
Looks good to me! :shipit:

**Pull request merged**